src/database/world_graph.lua

```diff
@@ -174,7 +174,12 @@ function upperSandZone:new(worldGraph)
     return self.world.regions.arthur:canAccess(items)
   end
 
-  self.locations.curly.requirements = function(self, items) return _has(items, "polarStar") end
+  self.locations.curly.requirements = function(self, items)
+    return _has(items, "polarStar") and _has(items, "weaponBoss")
+  end
```

Comment:
this specific location's requirements can stay the same as they were - if you have `polarStar`, you're guaranteed to have `weaponBoss`